src/open_inwoner/components/templatetags/kvk_tags.py

```diff
@@ -0,0 +1,180 @@
+import json
+
+from django import template
+from django.utils.html import escape
+from django.utils.safestring import mark_safe
+
+register = template.Library()
+
+
+@register.simple_tag
+def react_branch_data(branches, selected_id=None):
+    """
+    Convert Django branch data to React-compatible JSON format
+    This matches exactly what SelectComboBoxModule expects
+
+    This function creates JSON data that gets embedded in HTML and read by React.
+    The 'selected_id' corresponds to the branch_number field that Django expects
+    when the form is submitted via POST request.
+    """
+    items = []
+    for branch in branches:
+        # Handle the "entire company" case (empty vestigingsnummer)
+        # This becomes the 'id' that React stores and sends back as 'branch_number'
+        branch_id = branch.get("vestigingsnummer", "") or "entire-company"
+
+        # Build structured additional info for multi-line display in React dropdown
+        vestiging_info = ""
+        address_info = ""
+        city_info = ""
+        rechtspersoon_info = ""
+
+        if branch.get("vestigingsnummer"):
+            vestiging_info = f"Vestiging: {branch['vestigingsnummer']}"
+            if branch.get("type") == "hoofdvestiging":
+                vestiging_info += " (Hoofdvestiging)"
+        else:
+            # Show "(Rechtspersoon)" as separate line for entire company option
+            # This handles both explicit rechtspersoon entries AND generated entire-company options
+            rechtspersoon_info = "(Rechtspersoon)"
```

Comment:
```suggestion
        if branch_id == "rechtspersoon":
            rechtspersoon_info = "(Rechtspersoon)"
        else:
            vestiging_info = f"Vestiging: {branch['vestigingsnummer']}"
            if branch.get("type") == "hoofdvestiging":
                vestiging_info += " (Hoofdvestiging)"
```

you can get the `branch_id` above, no need to `get` again here